ReactNativeClient/lib/components/note-body-viewer.js

```diff
@@ -203,6 +210,33 @@ class NoteBodyViewer extends Component {
 		return this.forceUpdate_;
 	}
 
+	async onResourceLongPress(resourceId) {
+		const resource = await Resource.load(resourceId);
+		const name = resource.title ? resource.title : resource.file_name;
+
+		const action = await dialogs.pop(this, name, [
+			{ text: _('Share'), id: 'share' },
+		]);
+
+		if (action === 'share') {
+			const file_name = resource.file_name ?
+				`${resource.file_name}.${resource.file_extension}` :
+				resource.title;
+			const targetPath = `${Setting.value('resourceDir')}/${file_name}`;
+
+			await shim.fsDriver().copy(Resource.fullPath(resource), targetPath);
+
+			await Share.open({
```

Comment:
What happens in case of error here? Does the app crash or it just does nothing?

Reply:
added a catch block just in case